### DISCUSSION TITLE
Option to embedding frontend application

### DISCUSSION THREAD

**Discussion by felipesere** [loco-rs/loco#348] [General]
👋 I know there is some discussion about picking the right frontend framework, but I wanted to side-step it and talk about
embedding the files directly into my final binary.
While I'm happy to have external config to my binary, I consider the frontend to be a tight part of it and I'd like to not have to think about how to ship it along my app. Docker technically helps, but there are also nice crates such as `rust-embed` that make producing a single binary easier.

At least so I thought!
At the moment, `loco` tightly wraps the `axum::Router` and I can't add a fallback_route of my own. 
The only path is through the static asset middleware...

Is there interest in maybe swapping how static assets are handled (i.e. embedding them) or opening up this avenue for customization (i.e. access the underlying `axum::Router`?

**Comment by jondot** [comment]
Did you check the `src/app.rs` hook integrations? It allows you to "plug into" various stages of an app's lifecycle and customize components. In `after_routes` you have access to a live Axum Router, so you can customize and add stuff to it.

**Reply by felipesere** [reply]
Oh sweet! No I had not looked at `after_routes` :)
